### PR TITLE
Suppport for Init Containers dashboard

### LINKED ISSUE #1909: Show initContainers amongst other containers
version 1.6

I would like to see the initContainers same like the regular containers, with environment variables, logs etc.

### PR DESCRIPTION
**Pull request by kenan435**
fixes #1909 
Dashboard will now display Init Containers in addition to normal containers.

### CODE REVIEW

File: src/app/frontend/pod/detail/init_containerinfo.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<kd-info-card>
+  <kd-info-card-header>[[Init Containers|Subtitle at the top of the container details.]]</kd-info-card-header>
```
Comment (cheld) [inline src/app/frontend/pod/detail/init_containerinfo.html]: It seems the title is the only difference between init-container and container. Can we refactor to reduce code duplication?

File: src/app/externs/backendapi.js [hunk 2]
```diff
@@ -776,8 +765,31 @@ backendApi.Pod;
...
+ *   args: Array<string>
+ * }}
+ */
+backendApi.InitContainer;
+
+/**
+ * @typedef {{
  *   objectMeta: !backendApi.ObjectMeta,
  *   typeMeta: !backendApi.TypeMeta,
+ *   initContainers: !Array<!backendApi.InitContainer>,
```
Comment (cheld) [inline src/app/externs/backendapi.js]: Do we need a new type?

File: src/app/backend/resource/pod/detail.go [hunk 3]
```diff
@@ -202,47 +205,50 @@ func isNotFoundError(err error) bool {
...
 }
 
-func toPodDetail(pod *v1.Pod, metrics []metric.Metric, configMaps *v1.ConfigMapList,
-	secrets *v1.SecretList, controller owner.ResourceOwner,
-	events *common.EventList) PodDetail {
-
+func extractContainerInfo(containerList []v1.Container, pod *v1.Pod, configMaps *v1.ConfigMapList, secrets *v1.SecretList) []Container {
 	containers := make([]Container, 0)
-	for _, container := range pod.Spec.Containers {
+	for _, c := range containerList {
```
Comment (cheld) [inline src/app/backend/resource/pod/detail.go]: nit: please do not use single character variable names (exception if you want to indicate to the reader that the variable is unimportant and can be ignored, e.g. iterator i)

**Comment by kenan435** [comment]
Suggestions implemented. PTAL @floreks @cheld

File: src/app/frontend/pod/detail/containerinfo_component.js [hunk 4]
```diff
@@ -61,15 +64,30 @@ export default class ContainerInfoController {
     return this.state_.href(
         logsStateName, new LogsStateParams(this.namespace, this.podName, container.name));
   }
+
+  /**
+   * @return {string}
+   * @export
+   */
+  getTitle() {
```
Comment (cheld) [inline src/app/frontend/pod/detail/containerinfo_component.js]: this method seems not to be used, with the current transclude approach

File: src/app/frontend/pod/detail/containerinfo_component.js [hunk 5]
```diff
@@ -39,6 +39,9 @@ export default class ContainerInfoController {
     /** @export {string} Initialized from a binding. */
     this.namespace;
 
+    /** @export {string} Initialized from a binding. */
+    this.isInitContainer;
```
Comment (cheld) [inline src/app/frontend/pod/detail/containerinfo_component.js]: this variable seems not to be used in the current implementation with transclude

**Comment by floreks** [comment]
I'll try to review it later. I have actually suggested transclude approach. It allows to easy override messages and works nicely with our translations. I'd maybe change tag name of the transclude but overall LGTM.

**Comment by kenan435** [comment]
PTAL @floreks @cheld Sorry, did not catch the dead code on Friday trying to get home.

**Comment by cheld** [comment]
LGTM

**Pull request merged**